feature_engine/timeseries/forecasting/lag_features.py

```diff
@@ -0,0 +1,292 @@
+            The dataframe with the original variables plus the new variables.
+        """
+        # Check method fit has been called
+        check_is_fitted(self)
+
+        # check if 'X' is a dataframe
+        X = _is_dataframe(X)
+
+        # Check if input data contains same number of columns as dataframe used to fit.
+        _check_input_matches_training_df(X, self.n_features_in_)
+
+        # check if dataset contains na
+        if self.missing_values == "raise":
+            _check_contains_na(X, self.variables_)
+            _check_contains_inf(X, self.variables_)
+
+        if isinstance(self.freq, list):
+            df_ls = []
+            for fr in self.freq:
+                tmp = X[self.variables_].shift(
+                    freq=fr,
+                    axis=0,
+                )
+                df_ls.append(tmp)
+            tmp = pd.concat(df_ls, axis=1)
+
+        elif isinstance(self.periods, list):
+            df_ls = []
+            for pr in self.periods:
+                tmp = X[self.variables_].shift(
+                    periods=pr,
+                    axis=0,
+                )
+                df_ls.append(tmp)
+            tmp = pd.concat(df_ls, axis=1)
+
+        else:
+            tmp = X[self.variables_].shift(
+                periods=self.periods,
+                freq=self.freq,
+                axis=0,
+            )
+
+        tmp.columns = self.get_feature_names_out()
+
+        X = X.merge(tmp, left_index=True, right_index=True, how="left")
+
+        if self.drop_original:
+            X = X.drop(self.variables_, axis=1)
+
+        return X
+
+    def get_feature_names_out(self, input_features=None):
+        """Get output feature names for transformation.
+
+        Parameters
+        ----------
+        input_features : array-like of str or None, default=None
+            Input features. If `input_features` is `None` then the names for all the
+            created features is returned. Alternatively, only the names for the
+            indicated features is returned.
+
+        Returns
+        -------
+        feature_names_out : list of str objects
+            Transformed feature names.
+        """
+        check_is_fitted(self)
+
+        # variable names will be created just for input_features.
+        if input_features is None:
+            input_features = self.variables_
+
+        if isinstance(self.freq, list):
+            feature_names = [
+                str(feature) + f"_lag_{fr}"
+                for fr in self.freq
+                for feature in input_features
+            ]
+        elif self.freq is not None:
+            feature_names = [
+                str(feature) + f"_lag_{self.freq}" for feature in input_features
+            ]
+        elif isinstance(self.periods, list):
+            feature_names = [
+                str(feature) + f"_lag_{pr}"
+                for pr in self.perdiods
```

Comment:
Typo here which raises an AttributeError. Should be `periods` rather than `perdiods`.

Reply:
Also suggests that tests are needed where the input arguments are lists and combinations of lists and ints.

Reply:
I have corrected the typo in this PR: https://github.com/feature-engine/feature_engine/pull/370

However, we still probably need to add some additional tests as per the comment above.